cmd/termination-handler/main.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	"k8s.io/klog"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/termination"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/version"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+func main() {
+	var printVersion bool
+	flag.BoolVar(&printVersion, "version", false, "print version and exit")
+
+	klog.InitFlags(nil)
+	pollInterval := flag.Duration("poll-interval", 5*time.Second, "interval at which termination notice endpoint should be checked")
+	nodeName := flag.String("node-name", "", "name of the node that the termination handler is running on")
```

Comment:
how's the nodeName expected to be set?

Reply:
wouldn't we need to use an env variable that gets set when running the pod?
```
      env:
        - name: MY_NODE_NAME
          valueFrom:
            fieldRef:
              fieldPath: spec.nodeName
```

Reply:
I was expecting it to come from the downwards API as you've explained, then in the args all you need is:

```
args:
- --node-name=$(MY_NODE_NAME)
```

I'm not sure if there's benefit to doing this as a flag or whether we should just read a known ENV VAR, my original thought was that a flag not being set or being misspelt can be explained during flag parse, but we could achieve something similar ourselves easily for the env var, do you have a preference?

Reply:
flag seems good to me.